app/js/translations.js

```diff
@@ -35,24 +43,31 @@ module.exports = {
 
       // Unless setLng query param is set, use subdomain lang
       if (!req.originalUrl.includes('setLng') && lang != null) {
-        req.i18n.setLng(lang)
+        req.i18n.changeLanguage(lang)
       }
 
-      // If the set language (req.lng) is different from the language
-      // automatically determined by the i18next middleware (req.language), then
-      // set flag which will show a banner offering to switch to the appropriate
-      // library
-      if (req.language !== req.lng) {
-        req.showUserOtherLng = req.language
+      // If the set language is different from the language detection (based on
+      // the Accept-Language header), then set flag which will show a banner
+      // offering to switch to the appropriate library
+      const detectedLanguage = langDetector.detect(req, res)
+      if (req.language !== detectedLanguage) {
+        req.showUserOtherLng = detectedLanguage
       }
 
       next()
     }
 
-    const expressMiddleware = i18n.handle
+    const expressMiddleware = middleware.handle(i18n)
+
+    function backwardsCompatExpressMiddleware(req, res, next) {
+      req.i18n.translate = req.i18n.t
```

Comment:
I'm unsure if this is a good idea or not, but would greatly reduce the diff on the web side. Will revisit on Monday.

Reply:
Let's move this into line 56 -- that should require no changes in web (other than the package-lock)?

Reply:
I did try this on Friday, but was too distracted by the heat to get it working 😅  It's now done in https://github.com/overleaf/translations/pull/31/commits/3ec5127fd4e2d98dfb007f9acbd4e627d5e88e98.